### ISSUE TITLE
Success Criterion 2.5.2 - Pointer Cancellation - Level A

### ISSUE THREAD

**Issue by JJdeGroot** [w3c/matf#38]
WCAG2ICT guidance: https://www.w3.org/TR/wcag2ict-22/#pointer-cancellation

Share your thoughts for applying to mobile apps as a comment below.

**Comment by jha11y** [comment]
<details>
<summary>Notes from 11 September, 2024 Meeting:</summary>

[Meeting Minutes 11 September](https://www.w3.org/2024/09/11-matf-minutes.html#t02)

ACTION: Propose language to keep first 2 notes and add others as needed julianmka

Illai: [note#6](https://github.com/w3c/note/issues/6) is a good note to have on 2.5.2.

julianmka: we might want to keep [note#5](https://github.com/w3c/note/issues/5) on 2.5.2

Jamie: we need to redefine user agent so note 5 would need to change

Joe_Humbert: user agents mostly apply to web, but not on mobile

Joe_Humbert: need to recheck if the user agents has been defined in [note#5](https://github.com/w3c/note/issues/5) for 2.5.2

ACTION: Joe_Humbert to check with WCAG2ICT to see why they added [note#5](https://github.com/w3c/note/issues/5) on 2.5.2

</details>

**Comment by shoobe01** [comment]
Two cases I have encountered where down (and movement...) perform actions that cannot be obviously changed to work on up or have aborts such as gesture-away: 
- Gesture (move a carousel...)
- Drawing tools

Would these each be considered Essential or is there a way to distinctly abort such inputs I haven't considered so they could also be covered in examples to assure better compliance? (I do not consider manually reversing like gesture, then gesture back, to be an abort command myself).

**Comment by shoobe01** [comment]
Also noticed another this past week: the slider input control (a native mobile form element) has no up cancel. Unlike all other form elements, moving off the element then up has no impact. The down tap or drag instantly and persistently changes the input. 

But... I may not get the scope of work. Should we simply accept OS level decisions and have exceptions to them always, or push for best behavior? Without an exception, I'd find it hard to call a slider acting on down Essential, so to be fully compliant, devs will need to build a workaround (or we hope it pushes OS makers to be better).

**Comment by detlevhfischer** [comment]
> Should we simply accept OS level decisions and have exceptions to them always, or push for best behavior? 

I think exempting OS level patterns makes sense since users will know and even expect this behavour.

**Comment by shoobe01** [comment]
May be a philosophical/scope discussion I am not aware of yet since new: I have deigned several browsers, worked with people on OS design, and know they are all done like all app / site design; not infalliable and they change their minds so things aren't fixed. 
So I tend to say all standards should be absolute, to be aspirational. But am I wrong, we give a pass to OSs and have our guidelines but also "if OS does otherwise, that's also compliant"? If so, do we explicitly say that anywhere? If not, should we define this somewhere?

**Comment by shoobe01** [comment]
How I read the discussion from the 27 aug meeting: Allowing the "Each layer is responsible for its own pointer actions only" exception (in Note 1 above as well as the Note 6 elsewhere discussed in the meeting) allows designers/devs to give a pass to non-compliant OS level interactions (although as I read it doesn't _require_ accepting lower layer non-compliance... which makes me happy). 

So since we already have this in Note 1 above, no need to edit it further. 

I would want to talk further about Examples of essential functionality. Does it only have to be input or other system requirements? Mine has always been drawing tools (I have had to design a few sketching type tools, a11y of them has come up in review). On down, you draw. It can be aborted, undone, etc, but not with the pointer. When in drawing mode, you are drawing on down. Is there a more general way to talk about interactions intrinsically performed on down that could be added to the Example or a Note?

**Comment by JJdeGroot** [comment]
Discussed in MATF meeting on August 20.

<details>
<summary>20 August 2025</summary>

[Source](https://www.w3.org/2025/08/20-matf-minutes.html#953a)

~~~text
Joe_Humbert summarized 2.5.2 Pointer Cancellation discussion including comments from Detlev and Steven discussing OS behaviors

Joe_Humbert In WCAG there are a lot of exceptions for OS level or browser level things that developers cannot change, /If the OS is doing something and the developers can;t change because it I s part of the technology the expectation would not be that the developers and would not be a violation of WCAG

Joe_Humbert Steven brought up that OS changes, in essence not wanting to exempt OS level items

pauljadam does not see any exception in the wording of the SC itself, do we need this?

Joe_Humbert JJ was thinking about the essential behavior of the OS

pauljadam toggles are examples in iOS that may not be acting as expected such as undoing, but not sure if that fails if you can untoggle?

rachaely Can we discuss the examples of things that would be considered exempt ?

pauljadam not sure we really need to add language about this

Joe_Humbert Looking at Note 2 specifically of WCAG2ICT

pauljadam maybe like a piano app where you need to play on the down event?

rachaely or maybe the pull-down menu or Home Screen app behavior

pauljadam devs don't have control over those items

Tim the essential part seems more about the function, like the piano example, and not necessarily the essential behavior of the OS, that;s code not necessarily a function, so I'm hesitant to think of "essential" as "operating system"

Joe_Humbert if we can't think of things this may be moot

Jamie we might want to consider a phrasing at a higher level like "limitations of the technology" that is more general as that can address the concerns in the Github issue from Steven while not chasing every version of an OS

Joe_Humbert This SC has a lot of Notes, any comments from folks about them? We are ignoring Closed Functionality notes

rachaely points out that Note 6 explicitly added applicability of the OS so that's the exact opposite of what we just discussed.

Joe_Humbert no other notes discussed
~~~
</details>

**Comment by shoobe01** [comment]
Thought on examples for this. Should the only example be an exception? Should we have maybe one normal one to clarify up-abort for typical devs (you'd be surprised how many don't get up vs down), and one for the works-on-down case but we mention the concept of Undo, as separate from abort: 

Example 1
The user presses (down) on a button, then decides they do not want to make that selection, so can drag the pointer outside the button. On up, nothing happens; the button action has been aborted. 

Example 2 
A tool that draws on down cannot be aborted by up gesture as any movement records more drawing. Undo functions should be provided as separate functions.

**Comment by JJdeGroot** [comment]
Discussed in MATF meeting on 29 October 2025.

<details>
<summary>Minutes for 29 October 2025</summary>

[Source](https://www.w3.org/2025/10/29-matf-minutes.html#0e69)

~~~text
<JJ> https://github.com/w3c/matf/pull/267/files

<JJ> https://github.com/w3c/matf/pull/267/commits/7549f847dbcf1b8642ecf088858d7fc328e3a995

Joe_Humbert to review: Go to view files and then "submit review"

shoobe01 what if the app has a feature instead of is that kind of app - i.e. there is a drawing component as opposed to a drawing app

ACTION: For 2.5.2 replace "apps/applications" with "functionality" to apply to broader type of applications

Joe_Humbert This is why I am concerned about approving, because I have approval access and it get's merged

<Rob-w> I approve in principle, but can't hit approve on the machine I'm currently using.
~~~
</details>

- ACTION: For 2.5.2 replace "apps/applications" with "functionality" to apply to broader type of applications
